Allow pounce with a given input (and some slight lua refactoring)

IndianBoy42 opened the pull request:
- allow pounce with predefined input
- use tbl_extend for opts
- Move init to lua, Pounce command takes an argument

IndianBoy42 committed:
allow pounce with predefined input

IndianBoy42 committed:
Move init to lua, Pounce command takes an argument

.

IndianBoy42 committed:
use tbl_extend for opts

rlane commented on lua/pounce.lua:
```diff
@@ -54,9 +125,31 @@ local function calculate_proximity_bonus(cursor_line, cursor_col, match_line, ma
+function M.setup(opts)
+  M.config(opts)
+
+  for hl, spec in pairs(default_hl) do
```
I recently moved this to an autocmd so that colorschemes wouldn't reset it. Can the same be done in Lua? I also don't mind keeping the highlights in vimscript.

rlane commented:
Could you also update the readme?

IndianBoy42 committed:
Update readme, slightly improve the commands, properly init highlights

IndianBoy42 committed:
Pounce preview

IndianBoy42 commented:
I have updated the readme, it covers in somewhat basic terms, please tell me if its not clear. I also wasn't sure how many examples I should add about things you might want to use it for

I also added the PounceReg and PounceExpand commands since I think those cover the most common use cases for preinitialized input.

Finally I added support for the new command preview feature so while typing in the cmdline after Pounce it will actually do the highlighting and key labelling

IndianBoy42 committed:
More examples

rlane commented:
Thanks!
